generator_process/models/checkpoint.py

```diff
@@ -0,0 +1,9 @@
+from dataclasses import dataclass
+
+from .model_config import ModelConfig
+
+
+@dataclass
+class Checkpoint:
+    path: str
+    config: ModelConfig | str | None
```

Comment:
I had to add a `__hash__` function for this to work:
```suggestion
    config: ModelConfig | str | None
    
    def __hash__(self):
        return hash((self.path, self.config))
```
Otherwise, an `unhashable type` error occurred [here in the `cache_check` decorator](https://github.com/carson-katri/dream-textures/pull/703/files#diff-cbf852790591d9974d18a3d84f89f66fc64e3803ac294f17f5c1786838ece249R40)

Reply:
I opted for making the class frozen to allow it to generate the hash function. Seems best this way since it is used as a key.